numba/np/arrayobj.py

```diff
@@ -1208,7 +1235,6 @@ def maybe_copy_source(context, builder, use_copy,
             builder.store(builder.load(src_ptr), dest_ptr)
 
     def src_getitem(source_indices):
-        assert len(source_indices) == srcty.ndim
```

Comment:
Should this be checking something like `len(source_indices) == srcty.ndim + len(self.newaxes)` or is this sort of invariant now entirely lost?

Reply:
I removed it because during that function call we do not have the information regarding the number of `newaxes`/`None`s, since that is being unpacked/processed while calling the `Indexer`'s constructor which happens much later after this call. So at this specific line, yeah we can't do it unless we unpack the indices specifically for this `assert`.

Reply:
Ah I see, thanks for the explanation. I think it's ok to leave this as-is then.